internal/server/boltdbstate/config_source.go

```diff
@@ -68,11 +72,21 @@ func (s *State) configSourceSet(
 	memTxn *memdb.Txn,
 	value *pb.ConfigSource,
 ) error {
-	id := s.configSourceId(value)
+	// The scope and type of a config source is used to establish a unique record
+	// in the config sources table.
+	idHash, err := hashstructure.Hash(map[string]interface{}{
+		"scope":     value.Scope,
+		"type":      value.Type,
+		"workspace": value.Workspace,
+	}, hashstructure.FormatV2, nil)
+	if err != nil {
+		return err
+	}
+
+	id := s.configSourceId(idHash)
```

Comment:
Note for later readers, there is a note on configSourceId that it's only used for in memory indexes, so we aren't invalidating anything by changing the hash calculate algorithm.